### PR TITLE
Add log for vault corruption

### PR DESCRIPTION
**Pull request by gantunesr**

### CODE REVIEW

**Comment by owencraston** [comment]
This is going to be so helpful!

File: app/core/Analytics/Analytics.js [hunk 1]
```diff
@@ -400,7 +404,34 @@ class Analytics {
       default:
         RCTAnalytics.setUserProfileProperty(
           USER_PROFILE_PROPERTY.AUTHENTICATION_TYPE,
-          AUTHENTICATION_TYPE.UNKOWN,
+          AUTHENTICATION_TYPE.UNKNOWN,
```
Comment (owencraston) [inline app/core/Analytics/Analytics.js:407]: this was a typo

File: app/core/Analytics/Analytics.js [hunk 2]
```diff
@@ -400,7 +404,34 @@ class Analytics {
...
+        );
+    }
+  };
+
+  /**
+   * Apply Storage Type Property
+   *
+   * @param {string} property - A string representing the storage solution of the user. One of biometrics, AsyncStorage, FilesystemStorage or unknown
+   */
+  applyUserStorageTypeProperty = (property) => {
```
Comment (owencraston) [inline app/core/Analytics/Analytics.js:417]: For the sake of simplicity I opted to create a new method to handle this specific property

File: app/core/Analytics/Analytics.js [hunk 3]
```diff
@@ -574,8 +605,11 @@ export default {
   trackEvent(event, anonymously) {
     return instance && instance.trackEvent(event, anonymously);
   },
-  applyUserProperty(property) {
-    return instance && instance.applyUserProperty(property);
+  applyUserAuthenticationTypeProperty(property) {
```
Comment (owencraston) [inline app/core/Analytics/Analytics.js:608]: Since these methods handle specific properties I named them accordingly

File: bitrise.yml [hunk 4]
```diff
@@ -323,10 +323,10 @@ app:
           PROJECT_LOCATION_IOS: ios
         - opts:
               is_expand: false
-          VERSION_NAME: 5.3.0
+          VERSION_NAME: 5.3.333
```
Comment (owencraston) [inline bitrise.yml:326]: used for testing. Must be reverted before merge.

**Comment by owencraston** [comment]
I followed the same pattern as the the authentication type user property from [this PR](https://github.com/MetaMask/metamask-mobile/pull/3875)

**Review by sethkfman — APPROVED** [review]
LGTM

**Review by andrepimenta — APPROVED** [review]
LGTM!

**Review by sethkfman — APPROVED** [review]
LGTM

**Comment by owencraston** [comment]
after trying to set the user properties in this PR I opted to revert the changes and made [a separate issue](https://github.com/MetaMask/metamask-mobile/issues/4733) to track this work

**Pull request merged**